content/howto/extensibility/how-to-create-a-native-pluggable-widget.md

```diff
@@ -0,0 +1,1344 @@
+
+- Generate a widget structure
+- Create...
+- Build...
+- Configure...
+
+**Are you in a hurry?**
+
+Clone this [code sample](https://github.com/mendix/native-group-box-pluggable-widget-sample) from GitHub with all group box features from this how-to already implemented.
+
+## 2 Prerequisites
+
+Before starting this how-to, make sure you have completed the following prerequisites:
+
+- Install Mendix Studio Pro 8.2.2
+- Install Long Term Support (LTS) v10.16.3 or any higher version of [Node.js](https://nodejs.org)
+- Install latest [Yeoman](https://yeoman.io/) with the following command:
+
+```shell
+npm install -g yo
+```
+
+- Install Mendix' Pluggable Widget Generator v8.2.0 for Yeoman with the following command:
+
+```shell
+npm install -g @mendix/generator-widget@8.2.0
+```
+
+- Install an integrated development environment (IDE) of your choice (Mendix recommends [Microsoft Visual Studio Code](https://code.visualstudio.com/))
+- Have a basic understanding of [TypeScript](https://www.typescriptlang.org/) and [React](https://reactjs.org)
+
+## 3 {Title of Section [Use Present Participle Verb/Gerund]}
+
+### TEMP: Structure
+
+- Setting up pluggable widget project directory
+
+  - Using generator
+    - Explain generated code
+  - Creating a test project
+  - Put test project in dist folder
+  - Run sample code from generator in test project
+
+- Transform sample code in groupbox code
+  - Create display component with basic look and header caption
+    - iOS: round corners
+    - Android: sharp corners
+  - Change glue component to passthrough header caption
+  - Add containment
+  - Add toggable functionality
+    - Touchable component (dynamic component)
+  - Add icon property for expand and collapse
+  - Add collapsible property
+  - Add default styling from modeler
+  - Create design property
+
+To {do this task}, follow these steps:
+
+1. {Step 1}
+2. {Step 2}
+
+![](attachments/{sub-folder with same name as doc file}/{image filename}.png)
+
+### 3.1 Scaffolding pluggable widget project
+
+The Pluggable Widget Generator is the quickest way to start developing a pluggable widget. When you run this generator with Yeoman, Yeoman will scaffold your project folder with a folder structure and certain files recommended by Mendix.
+
+To scaffold your project folder for the group box widget, open up a terminal, navigate to the folder where you want to store your project and start the generator by executing the following command:
+
+```shell
+yo @mendix/widget GroupBox
+```
+
+The generator will ask you for some input during setup. Please provide the following information:
+
+1. Widget name: _{Your widget name}_
+2. Widget description: _{Your widget description}_
+3. Organization name: _{Your organization name}_
+4. Copyright: _{Your copyright date}_
+5. License: _{Your license}_
+6. Initial version: _{Your initial version number}_
+7. Author: _{Your author name}_
+8. Mendix Project path: _./test/MxTestProject/_
```

Comment:
@diego is there is no content in the project path is should not generate the content there. as you can not create a project in a non empty folder.